thirdparty/build-thirdparty.sh

```diff
@@ -506,7 +506,7 @@ build_rocksdb() {
 build_sasl() {
     check_if_source_exist $SASL_SOURCE
     cd $TP_SOURCE_DIR/$SASL_SOURCE
-    CFLAGS= ./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
+    CFLAGS="" LDFLAGS="-L${TP_LIB_DIR}" ./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
```

Comment:
the correct way to specify customized package location in autoconf, is
```
./configure --with-openssl=$TP_INSTALL_DIR
```

here it is `./autogen.sh --with-openssl=$TP_INSTALL_DIR`

Reply:
right, but LDFLAGS is more general, what if sasl build links other libraries?

Reply:
then explicitly link each of the libraries. LDFLAGS just provides additional path to search, doesn't guarantee the search order, or even static or shared library, and it doesn't control the include path as well.

Right now we have too many similar issues in thirdparty build, because of CMAKE/autoconf implicit searching and link, the only way to make sure everything get to the right libraries, is to have a pure clean env.